website/src/layouts/base/header/OffCanvasOverlay.tsx

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useState } from 'react';
+
+type Props = {
+    onClick?: () => void;
+    className?: string;
+};
+
+export function OffCanvasOverlay({ onClick, className }: Props): JSX.Element {
+    return <div className={`fixed inset-0 z-20 bg-gray-800 bg-opacity-50 ${className ?? ''}`} onClick={onClick} />;
+}
+
+export const useOffCanvas = (initialIsOpen = false) => {
+    const [isOpen, setOpen] = useState(initialIsOpen);
+
+    const open = useCallback(() => {
+        document.body.style.overflow = 'hidden'; // This makes the background not scrollable
```

Comment:
How important is that feature actually? It seems overly complicated to me just for the sake of making the background not scrollable.

An alternative solution would be to make the open hamburger menu use the full page, then the problem whether the background is scrollable or not simply disappears.